pde/grids/operators/cartesian.py

```diff
@@ -984,45 +997,67 @@ def divergence(arr: np.ndarray, out: np.ndarray) -> None:
         with np.errstate(all="ignore"):
             # some errors can happen for ghost cells
             for i in range(len(data_shape)):
-                out += ndimage.convolve1d(arr[i], [1, 0, -1], axis=i)[valid] * scale[i]
+                out += ndimage.correlate1d(arr[i], stencil, axis=i)[valid] * scale[i]
 
     return divergence
 
 
-def _make_divergence_numba_1d(grid: CartesianGrid) -> OperatorType:
+def _make_divergence_numba_1d(
+    grid: CartesianGrid, method: Literal["central", "forward", "backward"] = "central"
+) -> OperatorType:
     """make a 1d divergence operator using numba compilation
 
     Args:
         grid (:class:`~pde.grids.cartesian.CartesianGrid`):
             The grid for which the operator is created
+        method (str):
+            The method for calculating the derivative. Possible values are 'central',
+            'forward', and 'backward'.
 
     Returns:
         A function that can be applied to an array of values
     """
+    if method not in {"central", "forward", "backward"}:
```

Comment:
If you are using `Literal`, you don't need this check :-)

Reply:
I don't think I agree. Typing is optional in python, so a user could easily pass different values (for instance by mis-typing!) and those would not be caught at runtime. The check basically ensures that the right values were given at runtime, whereas the type hint helps developers with proper IDEs and type checking.

Reply:
are you sure? Can you give an example?

If you look into the docs of [Literal](https://typing.readthedocs.io/en/latest/spec/literal.html#literal-types) you can see that it is not optional, but you have to give a specified value.

But better double checking than not checking at all ;-)